test/core/sup_tree_core/periodic_log/reduction_log_builder_test.exs

```diff
@@ -3,9 +3,23 @@ defmodule AntikytheraCore.PeriodicLog.ReductionBuilderTest do
 
   @dir Path.join([__DIR__, "..", "..", "..", "..", "_build", "test", "log", "antikythera"]) |> Path.expand()
 
-  test "should exists" do
-    assert Process.whereis(ReductionBuilder) != nil
-    send(ReductionBuilder, :timeout)
+  test "should exist" do
+    pid = Process.whereis(ReductionBuilder)
+    assert pid != nil
     assert File.exists?(Path.join(@dir, "reduction.log.gz"))
+
+    state1 = :sys.get_state(pid)
+    build_state1 = state1.build_state
+    # NOTE: The below `accert` statement won't pass. The process probably left its initial state
+    # because it has been running from the beginning of the test.
+    # assert state1.log_state.empty?
+
+    send(ReductionBuilder, :timeout)
+
+    state2 = :sys.get_state(pid)
```

Comment:
`sys.get_state/1` is implemented using messages. So it seems to be guaranteed that logging is complete at this point.